hamilton/execution/graph_functions.py

```diff
@@ -285,6 +251,84 @@ def dfs_traverse(
     return computed
 
 
+# TODO: better function name
+def execute_lifecycle_for_node(
+    node_: node.Node,
+    adapter: LifecycleAdapterSet,
+    run_id: str,
+    task_id: str,
```

Comment:
these will need to have `__` prefixes. Since we don't want to clash with possible values in`**kwargs` argument below; e.g. most people will not name variables with a `__` prefix so I think we should be good here.

Reply:
My bad, I got confused with ```partial``` freezing them, but they can clash before that..

Reply:
No worries - functools.partial will actually swallow any error it seems:
e.g.
```python
>>> def foo(a, b, c, **kwargs):
     print(a,b,c,kwargs)
     return "hi"
 
>>> f = functools.partial(foo, a=1, b=2, c=3)
>>> f(a=5)  # this will just override it -- won't error
5 2 3 {}
'hi'

>>> f(d=3)
1 2 3 {'d': 3}
'hi'

```

Reply:
That is good to know, but also wth - somebody somewhere is probably having a big headache because of this...